### DISCUSSION TITLE
[BUG] --recursive bug, configs bug and code sign

### DISCUSSION THREAD

**Discussion by pauloswear** [dashingsoft/pyarmor#1361] [Q&A]
Hello Jondy. There is some bugs happening here.

1)
--recursive does not obfuscate all files., Actually, the behaviour is the same as using --exact

```
...\launcher>pyarmor gen -r launcher.py
INFO     Python 3.11.4
INFO     Pyarmor 8.2.9 (pro), **, **
INFO     Platform windows.x86
INFO     search inputs ...
INFO     find script launcher.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'windows.amd64'}
INFO     write dist\pyarmor_runtime_0000\pyarmor_runtime.pyd
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "launcher"
INFO     obfuscating file launcher.py
INFO     write dist\launcher.py
INFO     obfuscate scripts OK
```

2)
The .pyarmor folder that contains config it's not been updated when I use new configs (pyarmor cfg). It would be good if there's an option to remove folder after run CLI

3)
Pyarmor gen always include pyarmor mark in output file and we can't disable it.
Example: 
`Pyarmor 8.2.9 (pro), VAX, DATETIME`

Thanks

**Comment by jondy** [comment]
They're not bug, but work as designed, please check `pyarmor gen` and `pyarmor cfg` documentation to find what's Pyarmor doing.

**Reply by pauloswear** [reply]
pyarmor gen --recursive launcher.py should obfuscate launcher and other scripts that launcher imports as documented here https://pyarmor.readthedocs.io/en/stable/reference/man.html?highlight=recursive

It's not working

**Reply by pauloswear** [reply]
> INFO     Python 3.10.5
> INFO     Pyarmor 8.2.9 (pro), vax, **
> INFO     Platform windows.x86_64
> DEBUG    native platform windows.amd64
> DEBUG    home path: C:\Users\Paulo\.pyarmor
> DEBUG    command options: {'recursive': True, 'no_runtime': False, 'inputs': ['launcher.py']}
> DEBUG    install plugin: CodesignPlugin
> INFO     search inputs ...
> INFO     find script launcher.py
> INFO     find 1 top resources
> INFO     start to generate runtime files
> INFO     target platforms {'windows.amd64'}
> INFO     write dist\pyarmor\pyarmor_runtime.pyd
> INFO     patch runtime file
> DEBUG    patching runtime data at 515104
> DEBUG    patch runtime file OK
> INFO     generate runtime files OK
> INFO     start to obfuscate scripts
> INFO     process resource "launcher"
> INFO     obfuscating file launcher.py
> DEBUG    process inline marker
> DEBUG    parse script
> DEBUG    process co
> DEBUG    patch co
> INFO     write dist\launcher.py
> INFO     obfuscate scripts OK


as debug shows, recursive is enabled but it's not searching for other scripts that launcher imports

command options: {'recursive': True, 'no_runtime': False, 'inputs': ['launcher.py']}

**Comment by jondy** [comment]
`--recursive` only works for path (package), refer to the description of `pyarmor gen`
https://pyarmor.readthedocs.io/en/stable/reference/man.html#pyarmor-gen

It's different from Pyarmor 7.x

**Reply by pauloswear** [reply]
Thanks! I returned use python to my projects and now it will be useful.
What about code sign?

**Comment by pauloswear** [comment]
#UP 

Every script has this signature, don't want to send script with this. How to remove this: (Again, nothing on docs)
# Pyarmor 8.4.2 (pro), VAX, ***, 2023-11-01T15:16:51.450360
